File: 485_MaxConsecutiveOnes.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    var m = 0, t = 0;
    for (var i = 0; i < nums.length; ++i) {
        if (nums[i] === 1) {
            t += 1;
        } else {
            m = Math.max(m, t);
            t = 0;
        }
    }
    return Math.max(m, t);
};

File: 485.MaxConsecutiveOnes.java
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int result = 0;
        int temp = 0;
        for (int i = 0; i < nums.length; i += 1) {
            if (nums[i] == 1) {
                temp += 1;
            } else {
                if (temp > result) {
                    result = temp;
                }
                temp = 0;
            }
        }
        if (temp > result) {
            result = temp;
        }
        return result;
    }
}